src/hotspot/share/runtime/os.cpp

```diff
@@ -1037,7 +1042,12 @@ void os::print_hex_dump(outputStream* st, const_address start, const_address end
   // Print out the addresses as if we were starting from logical_start.
   while (p < end) {
     if (cols == 0) {
-      st->print(PTR_FORMAT ":   ", p2i(logical_p));
+      // highlight start of line if address of interest is located in the line
+      bool should_highlight = false;
+      if (highlight_address >= p && highlight_address < p + cols_per_line) should_highlight = true;
```

Comment:
Can be shortened, and should use the byte line size, not the size in printed units.

```suggestion
      const bool should_highlight = (highlight_address >= p && highlight_address < p + bytes_per_line);
```

This highlights a gap in your testing. You should also test with unitsizes other than 1.